example/lib/devices_list/devices_bloc.dart

```diff
@@ -86,13 +86,13 @@ class DevicesBloc {
       if (_locationPermissionStatus != PermissionStatus.granted) {
         return Future.error(Exception("Location permission not granted"));
       }
-    } else if (Platform.isIOS) {
-      /// On iOS the user will be asked to allow Bluetooth access by
-      /// the system. Once it is granted, the Bluetooth state will change from
-      /// UNKNOWN to POWERED_ON. The following function will block the thread
-      /// until it happens.
-      await _waitForBluetoothPoweredOn();
     }
+
+    /// On iOS the user will be asked to allow Bluetooth access by
+    /// the system. Once it is granted, the Bluetooth state will change from
+    /// UNKNOWN to POWERED_ON. The following function will block the thread
+    /// until it happens.
+    await _waitForBluetoothPoweredOn();
```

Comment:
I also think that it is not directly related to permissions. Permission is OK, the state of adapter is unknown. WDYT?

Reply:
I moved `_waitForBluetoothPoweredOn()` to the `init()` chain, so it is done as a separate step after checking permissions. I think it looks more clean now. Is it OK for you?